Framework/DataHandling/src/SaveReflectometryAscii.cpp

```diff
@@ -90,50 +109,93 @@ void SaveMFT::data() {
     outputval(points[i]);
     outputval(yData[i]);
     outputval(eData[i]);
-    if (m_ws->hasDx(0))
-      outputval(m_ws->dx(0)[i]);
+    if ((!m_ext.compare("custom") && getProperty("WriteResolution")) ||
+        (!m_ext.compare(".mft")) || (!m_ext.compare(".txt"))) {
+      if (m_ws->hasDx(0))
+        outputval(m_ws->dx(0)[i]);
+      else {
+        if (m_ext.compare(".mft"))
+          outputval(points[i] * ((points[1] + points[0]) / points[1]));
+      }
+    }
     m_file << '\n';
   }
 }
 
+/// Determine separator
+void SaveReflectometryAscii::separator() {
+  if (!m_ext.compare("custom")) {
+    const std::string sepOption = getProperty("Separator");
+    if (sepOption == "tab" || !m_ext.compare(".txt"))
+      m_sep = '\t';
+    if (sepOption == "comma") {
+      m_sep = ',';
+    } else if (sepOption == "space") {
+      m_sep = ' ';
+    }
+  }
+}
+
+/** Write string value
+ * @param write :: if true, write string
+ * @param s :: string
+ */
+bool SaveReflectometryAscii::writeString(bool write, std::string s) {
+  if (write) {
+    if (m_sep != '\0')
+      m_file << m_sep << s;
```

Comment:
I now don't use `\0` anymore since it did not mean anything. Instead, I initialized m_sep with the tab `\t` and decide via the file extension is separation is used.